crates/chia-consensus/src/merkle_tree.rs

```diff
@@ -77,93 +77,114 @@ impl MerkleSet {
         let mut values = Vec::<(u32, NodeType)>::new();
         let mut ops = vec![ParseOp::Node];
         let mut depth = 0;
+        let mut bits_stack: Vec<Vec<bool>> = Vec::new();
+        bits_stack.push(Vec::new());
 
         while let Some(op) = ops.pop() {
-            match op {
-                ParseOp::Node => {
-                    let mut b = [0; 1];
-                    proof.read_exact(&mut b).map_err(|_| SetError)?;
-
-                    match b[0] {
-                        EMPTY => {
-                            values.push((self.nodes_vec.len() as u32, NodeType::Empty));
-                            self.nodes_vec.push((ArrayTypes::Empty, BLANK));
-                        }
-                        TERMINAL => {
-                            let mut hash = [0; 32];
-                            proof.read_exact(&mut hash).map_err(|_| SetError)?;
-                            values.push((self.nodes_vec.len() as u32, NodeType::Term));
-                            self.nodes_vec.push((ArrayTypes::Leaf, hash));
-                        }
-                        TRUNCATED => {
-                            let mut hash = [0; 32];
-                            proof.read_exact(&mut hash).map_err(|_| SetError)?;
-                            values.push((self.nodes_vec.len() as u32, NodeType::Mid));
-                            self.nodes_vec.push((ArrayTypes::Truncated, hash));
-                        }
-                        MIDDLE => {
-                            if depth > 256 {
+            if let Some(bits) = bits_stack.pop() {
```

Comment:
I think it would be a lot nicer to say:
```suggestion
            let Some(bits) = bits_stack.pop() else {
                return Err(SetError);
            }
```

Reply:
it saves an indentation level for a lot of code